### ISSUE TITLE
peek_string/3 does not terminate when length > buffer size

### ISSUE THREAD

**Issue by wouterbeek** [SWI-Prolog/swipl-devel#255]
The following does not terminate:

```prolog
mwe :-
  setup_call_cleanup(
    open('example.txt', read, In),
    peek_string(In, 4096, _),
    close(In)
  ).
```

The reason for this is that the buffer size of stream `In` is 4,096 (obtained using `stream_property/2`).  The code does work for values <4,096.

Expected: `peek_string/3` throws an exception if `Length` is instantiated with an integer that is larger than the buffer size.

**Comment by JanWielemaker** [comment]
Doesn't reproduce.   In theory, peek_string/3 will enlarge the buffer size if it is insufficient.  Possibly the exact length and character encoding of `example.txt` matters for repeating non-termination.  Please attach the file and make explicit which encoding was used.

**Comment by wouterbeek** [comment]
Example file obtained through `curl -vL "http://aap.nl" > example.txt`:

```
<!DOCTYPE html>
<html lang="nl" dir="ltr" xmlns:dc="http://purl.org/dc/terms/" xmlns:og="http://ogp.me/ns#" xmlns:article="http://ogp.me/ns/article#" xmlns:book="http://ogp.me/ns/book#" xmlns:product="http://ogp.me/ns/product#" xmlns:profile="http://ogp.me/ns/profile#" xmlns:video="http://ogp.me/ns/video#">
<head>
  <meta charset="utf-8" />
<meta name="title" content="Geeft exotische zoogdieren weer een toekomst | Stichting AAP" />
<link rel="shortlink" href="https://www.aap.nl/nl" />
<link rel="canonical" href="https://www.aap.nl/nl" />
<meta name="description" content="AAP geeft apen en andere exotische zoogdieren weer een toekomst. We vangen dieren in nood op én pleiten voor betere wetten in heel Europa." />
<meta name="referrer" content="no-referrer-when-downgrade" />
<meta name="Generator" content="Drupal 8 (https://www.drupal.org)" />
<meta name="MobileOptimized" content="width" />
<meta name="HandheldFriendly" content="true" />
<meta name="viewport" content="width=device-width, initial-scale=1.0" />
<link rel="shortcut icon" href="/favicon.ico" type="image/vnd.microsoft.icon" />
<link rel="alternate" hreflang="nl" href="https://www.aap.nl/nl/node/3667" />
<link rel="alternate" hreflang="es" href="https://www.aap.nl/es/node/3667" />
<link rel="revision" href="/nl/node/3667" />

  <title>Geeft exotische zoogdieren weer een toekomst | Stichting AAP</title>
  <link rel="stylesheet" href="/uploads/css/css_cOwHzT0nOe4_c0RceSo8evUwOjMCoozx1BJoSpvby6w.css?oskgma" media="all" />
<link rel="stylesheet" href="/uploads/css/css_wH0sHhwqiiKuQrThHJiUDDncQxFu2Ez-pZeej9he68g.css?oskgma" media="all" />

  
<!--[if lte IE 8]>
<script src="/uploads/js/js_VtafjXmRvoUgAzqzYTA3Wrjkx9wcWhjP0G4ZnnqRamA.js"></script>
<![endif]-->


  
  <!-- Adform Tracking Code BEGIN -->
  <script type="text/javascript">
  var _adftrack = {
   pm: 306424,
   pagename: encodeURIComponent('Stichting AAP'),
   divider: encodeURIComponent('|')
  };
  (function(){var s=document.createElement('script');s.type='text/javascript';s.async=true;s.src='https://track.adform.net/serving/scripts/trackpoint/async/';var x = document.getElementsByTagName('script')[0];x.parentNode.insertBefore(s, x);})();
  </script>
  <noscript>
   <p style="margin:0;padding:0;border:0;">
   <img src="https://track.adform.net/Serving/TrackPoint/?306424=306424&amp;ADFPageName=Stichting AAP&amp;ADFdivider=|" width="1" height="1" alt="" />
   </p>
  </noscript>
  <!-- Adform Tracking Code END -->

  <!-- Start Visual Website Optimizer Asynchronous Code -->
  <script type='text/javascript'>
    var _vwo_code=(function(){
      var account_id=290705,
        settings_tolerance=2000,
        library_tolerance=2500,
        use_existing_jquery=false,
        /* DO NOT EDIT BELOW THIS LINE */
        f=false,d=document;return{use_existing_jquery:function(){return use_existing_jquery;},library_tolerance:function(){return library_tolerance;},finish:function(){if(!f){f=true;var a=d.getElementById('_vis_opt_path_hides');if(a)a.parentNode.removeChild(a);}},finished:function(){return f;},load:function(a){var b=d.createElement('script');b.src=a;b.type='text/javascript';b.innerText;b.onerror=function(){_vwo_code.finish();};d.getElementsByTagName('head')[0].appendChild(b);},init:function(){settings_timer=setTimeout('_vwo_code.finish()',settings_tolerance);var a=d.createElement('style'),b='body{opacity:0 !important;filter:alpha(opacity=0) !important;background:none !important;}',h=d.getElementsByTagName('head')[0];a.setAttribute('id','_vis_opt_path_hides');a.setAttribute('type','text/css');if(a.styleSheet)a.styleSheet.cssText=b;else a.appendChild(d.createTextNode(b));h.appendChild(a);this.load('//dev.visualwebsiteoptimizer.com/j.php?a='+account_id+'&u='+encodeURIComponent(d.URL)+'&r='+Math.random());return settings_timer;}};}());_vwo_settings_timer=_vwo_code.init();
  </script>
  <!-- End Visual Website Optimizer Asynchronous Code -->
</head>
<body>
<!-- Google Tag Manager -->
<noscript><iframe src="//www.googletagmanager.com/ns.html?id=GTM-T3G85F"
height="0" width="0" style="display:none;visibility:hidden"></iframe></noscript>
<script>(function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
'//www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);
})(window,document,'script','dataLayer','GTM-T3G85F');</script>
<!-- End Google Tag Manager -->

<a href="#main-content" class="visually-hidden focusable skip-link">Overslaan en naar de inhoud gaan</a>

<div class="page">

    <nav class="page__navbar navbar navbar-default" role="navigation" id="mainnav">
    <div class="container">
      <!-- Brand and toggle get grouped for better mobile display -->
      <div class="navbar-header">
        <button type="button" class="navbar-toggle" data-toggle="collapse" data-target="#navbar-content">
          <span class="sr-only">Toggle navigation</span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
        </button>
        <a class="navbar-brand" href="https://www.aap.nl/nl">
                      <img class="hidden-xs" src="/themes/aap/images/svg/logo-stichting-aap.svg" width="127"
                 height="177"
                 alt="Logo of Stichting AAP">
                    <img class="visible-xs" src="/themes/aap/images/logo-stichting-aap-2.png" width="80" height="27"
               alt="Logo of Stichting AAP">
        </a>
      </div>

      <!-- Collect the nav links, forms, and other content for toggling -->
      <div class="collapse navbar-collapse" id="navbar-content">
        <div class="navbar-topnav">
      <ul class="menu nav navbar-nav navbar-right">
	<li class="dropdown">
		<a href="#" class="dropdown-toggle" data-toggle="dropdown">Nederlands <span class="caret"></span></a><ul class="dropdown-menu" role="menu"><li hreflang="nl" data-drupal-link-system-path="&lt;front&gt;" class="nl is-active"><a href="/nl" class="language-link is-active" hreflang="nl" data-drupal-link-system-path="&lt;front&gt;">Nederlands</a></li><li hreflang="en" data-drupal-link-system-path="&lt;front&gt;" class="en"><a href="/en" class="language-link" hreflang="en" data-drupal-link-system-path="&lt;front&gt;">English</a></li><li hreflang="es" data-drupal-link-system-path="&lt;front&gt;" class="es"><a href="/es" class="language-link" hreflang="es" data-drupal-link-system-path="&lt;front&gt;">Español</a></li></ul>


	</li>
</ul>


        <ul class="menu nav navbar-nav navbar-right">
                      <li>
          <a href="/nl/over-ons-0" data-drupal-link-system-path="node/3314">Over ons</a>
                  </li>
                      <li>
          <a href="/nl/contact" data-drupal-link-system-path="node/2999">Contact</a>
                  </li>
          </ul>
  


  </div>


        <div class="navbar-mainnav">
      
<div class="navbar-mainnav-nav">
  
        <ul class="nav navbar-nav">
              <li>
          <a href="/nl/help-mee" data-drupal-link-system-path="node/3148">Help mee</a>
                  </li>
              <li>
          <a href="/nl/dit-doen-wij" data-drupal-link-system-path="what-we-do">Dit doen wij</a>
                  </li>
              <li>
          <a href="/nl/kom-kijken" data-drupal-link-system-path="node/1443">Kom kijken</a>
                  </li>
              <li>
          <a href="/nl/actueel" data-drupal-link-system-path="news-and-blogs">Actueel</a>
                  </li>
              <li>
          <a href="/nl/search" data-drupal-link-system-path="search">Zoeken</a>
                  </li>
          </ul>
  


<ul class="nav navbar-nav navbar-nav--social">
  <li><a href="https://www.facebook.com/StichtingAAP" target="_blank">Facebook</a></li>
</ul>

</div>

<div class="navbar-mainnav-buttons">
  
        <ul class="nav navbar-nav">
              <li>
          <a href="/nl/help-mee/adopteer-een-dier" data-drupal-link-system-path="help-mee/adopteer-een-dier">Adopteren</a>
                  </li>
              <li>
          <a href="/nl/help-mee/marria-donatie">Doneren</a>
                  </li>
          </ul>
  


</div>

  </div>

      </div>
      <!-- /.navbar-collapse -->
    </div>
  </nav>
  
      <nav class="page__tabs container">
        <div class="region-tabs">
          
      </div>

    </nav>
  
  <div class="page__body">
    <main class="page__main" role="main">
      <a id="main-content"></a>
            <div class="page__fluid-content">
        <div id="block-pageheader" class="block-pageheader">
  
    
              <div class="field-page__header">
              <div class="field-page__header__item">
<a id="lane-878"></a>
<div class="lane lane--medium paragraph-visual-with-ctas paragraph-visual-with-ctas--bottom">
                <div class="paragraph-visual-with-ctas__background-image-portrait">
              <div class="paragraph-visual-with-ctas__background-image-portrait-item">  <img src="/uploads/styles/visual_sm/public/background/2017-07/Marria%20mobiel.jpg?itok=W4_yBtL1" width="1233" height="1081" alt="Portretfoto van Chimpansee Marria" />


</div>
        </div>
        <div class="paragraph-visual-with-ctas__background-image">
              <div class="paragraph-visual-with-ctas__background-image-item">    <picture>
                <!--[if IE 9]><video style="display: none;"><![endif]-->
              <source srcset="/uploads/styles/visual_lg/public/background/2017-07/Marria.jpg?itok=8PssegjE 1x" media="all and (min-width: 1200px)" type="image/jpeg"/>
              <source srcset="/uploads/styles/visual_md/public/background/2017-07/Marria.jpg?itok=aQ1UvWYU 1x" media="all and (min-width: 992px)" type="image/jpeg"/>
              <source srcset="/uploads/styles/visual_sm/public/background/2017-07/Marria.jpg?itok=Tpwj3_jp 1x" media="all and (min-width: 768px)" type="image/jpeg"/>
              <source srcset="/uploads/styles/visual_xs/public/background/2017-07/Marria.jpg?itok=U4Mp47dX 1x" media="(min-width: 0em)" type="image/jpeg"/>
            <!--[if IE 9]></video><![endif]-->
            <img srcset="/uploads/styles/visual_md/public/background/2017-07/Marria.jpg?itok=aQ1UvWYU" alt="Portretfoto van Chimpansee Marria" />

  </picture>

</div>
        </div>
        <div class="paragraph-visual-with-ctas__body">
              <div class="paragraph-visual-with-ctas__body-item"><h2 class="heading-hero--small">Geef Marria weer een toekomst</h2>

<p> </p>

<p> </p></div>
        </div>
        <div class="paragraph-visual-with-ctas__ctas">
      <div class="paragraph-visual-with-ctas__ctas-items">
              <div class="paragraph-visual-with-ctas__ctas-item">
<a id="lane-876"></a>
<div class="paragraph-cta partial partial--blank">
              <div class="paragraph-cta__image">
              <div class="paragraph-cta__image-item">  <img src="/uploads/2017-07/phone.png" width="57" height="111" alt="Foto van een telefoon" />

</div>
        </div>
        <div class="paragraph-cta__title">
              <div class="paragraph-cta__title-item">Doneer je oude mobiel/cartridge</div>
        </div>
        <div class="paragraph-cta__link">
              <div class="paragraph-cta__link-item"><a href="/nl/marria" class="btn btn-secondary">Inleveren</a></div>
        </div>

  </div>
</div>
          <div class="paragraph-visual-with-ctas__ctas-item">
<a id="lane-875"></a>
<div class="paragraph-cta partial partial--blank">
              <div class="paragraph-cta__image">
              <div class="paragraph-cta__image-item">  <img src="/uploads/2017-07/airmiles.png" width="72" height="72" alt="Ail Miles Logo" />

</div>
        </div>
        <div class="paragraph-cta__title">
              <div class="paragraph-cta__title-item">Spaar Air Miles voor marria</div>
        </div>
        <div class="paragraph-cta__link">
              <div class="paragraph-cta__link-item"><a href="/nl/help-mee/air-miles-doneren-marria" class="btn btn-secondary">Sparen</a></div>
        </div>

  </div>
</div>
          <div class="paragraph-visual-with-ctas__ctas-item">
<a id="lane-877"></a>
<div class="paragraph-cta partial partial--blank">
              <div class="paragraph-cta__image">
              <div class="paragraph-cta__image-item">  <img src="/uploads/2017-07/ideal.png" width="71" height="62" alt="iDeal logo" />

</div>
        </div>
        <div class="paragraph-cta__title">
              <div class="paragraph-cta__title-item">Help Marria met een gift</div>
        </div>
        <div class="paragraph-cta__link">
              <div class="paragraph-cta__link-item"><a href="/nl/chimpansee-marria" class="btn btn-secondary">Doneren</a></div>
        </div>

  </div>
</div>
          </div>
  </div>

    <div class="lane__break"></div>
</div>
</div>
        </div>

  </div>

      </div>
      
      
      <div class="page__row page__row--first">
        <div class="container">
          <div class="row">
            <div class="col-md-12">
                <div class="page__content">  <div class="region-content">
            
<article role="article" class="node-lanepage-full node-lanepage-full--promoted full">

  
    

  
  <div class="node-lanepage-full__content full__content">
            <div class="node-lanepage-full__page__body">
              <div class="node-lanepage-full__page__body-item">
<a id="lane-871"></a>
<div class="lane lane--white paragraph-text">
                <div class="paragraph-text__body">
              <div class="paragraph-text__body-item"><h2 class="heading-xl">AAP geeft apen en andere exotische zoogdieren weer een toekomst</h2>

<p class="par-lg"><img alt="Illustratie van een aap en een klein zoogdier" data-entity-type="file" data-entity-uuid="74d6437f-e90b-4451-82dc-a9017ab024a2" src="/uploads/inline-images/animals_0_0.png" class="align-left" />In 2016 hebben we 235<strong> nieuwe dieren</strong> opgevangen en <strong>148<br />
dieren</strong> herplaatst naar een nieuw huis!</p>

<p class="par-xl"><a class="link-read-more" href="/nl/dit-doen-wij/2016-vogelvlucht" title="Successen van AAP">Al onze resultaten</a> <a class="link-read-more" href="/nl/dit-doen-wij" title="Het werk van AAP">Wat AAP nog meer doet</a></p></div>
        </div>

    <div class="lane__break"></div>
</div>
</div>
          <div class="node-lanepage-full__page__body-item">
<a id="lane-874"></a>
<div class="lane lane--blank paragraph-text-three-cols">
      

  <div class="paragraph-text-three-cols__col1">
            
              <div class="paragraph-text-three-cols__col1-item"><h2 class="heading-lg">Actueel</h2></div>
      

            
              <div class="paragraph-text-three-cols__col1-content-item">
<a id="lane-872"></a>
<div class="paragraph-partial-block partial partial--blank">
              <div class="paragraph-partial-block__block">
              <div class="paragraph-partial-block__block-item">  <div class="views-element-container block-news-latest" id="block-news-latest">
    
        
    <div><div class="view-actueel-promoted js-view-dom-id-7aa86958bf25e38373b6e0e10e095f26e21895cf8d812c994fd65dc2b476872c"><div class="view-actueel-promoted__content"><div class="view-actueel-promoted__row"><article role="article" class="node-news-promoted node-news-promoted--promoted"><h2 class="node-news-promoted__title"><a class="node-news-promoted__title-link" href="/nl/news/aflevering-7-van-marrias-vlog" rel="bookmark"><span>Aflevering #7 van Marria&#039;s vlog</span></a></h2><footer class="node-news-promoted__meta"><div class="node-news-promoted__submitted">wo, 19/07/2017</div></footer><div class="node-news-promoted__content"><a class="node-news-promoted__content-link" href="/nl/news/aflevering-7-van-marrias-vlog"><div class="node-news-promoted__image"><div class="node-news-promoted__image-item"><img src="/uploads/styles/crop_small/public/image/THUMBNAIL%20%237.jpg?itok=yoSvgiPW" width="600" height="400" alt="Marria&#039;s vlog, aflevering 7" title="Marria&#039;s vlog, aflevering 7" /></div></div><div class="node-news-promoted__body"><div class="node-news-promoted__body-item"><p>In aflevering #7 van Marria's vlog heeft Marria nogal 'bonje met de buren'. Want buiten zijn is heerlijk, maar Marria's buren, Buddy en Unyoro, zorgen soms wel voor overlast. Gelukkig hoeft ze...</p></div></div><span class="node-news-promoted__read-more">&nbsp;Lees verder</span></a></div></article></div></div><div class="more-link"><a href="/nl/actueel">Meer actualiteit</a></div></div></div>
  </div>
</div>
        </div>

  </div>
</div>
      

  </div>
  <div class="paragraph-text-three-cols__col2 paragraph-text-three-cols__col2--beige">
            
              <div class="paragraph-text-three-cols__col2-item"><h2 class="heading-lg">Blijf op de hoogte</h2>

<p class="par-md"><strong>Schrijf je in en ontvang elke twee maanden de leukste verhalen over onze dieren.</strong></p>

<p class="par-md">Mis niks! De leukste nieuwtjes en blogs in je inbox. Zo lees je welke dieren we opvangen, welke projecten onze focus hebben en wat onze doelen zijn voor korte en lange termijn.</p></div>
      

            
              <div class="paragraph-text-three-cols__col2-content-item">
<a id="lane-873"></a>
<div class="paragraph-partial-block partial partial--blank">
              <div class="paragraph-partial-block__block">
              <div class="paragraph-partial-block__block-item"><div id="block-mailing-subscribe" class="block-mailing-subscribe">
  
    
      <p></p>
<form class="promail-form" data-drupal-selector="promail-form" action="https://webgenerator.createsend.com/t/r/s/ilnurl/" method="post" id="promail-form" accept-charset="UTF-8" role="form">
  <div class="form-group js-form-item js-form-type-textfield js-form-item-cm-ilnurl-ilnurl form-item form-item-cm-ilnurl-ilnurl form-type-textfield form-item--no-label">
      <label for="edit-email" class="form-item__label visually-hidden js-form-required form-item__label--required">E-mailadres</label>
          <input data-drupal-selector="edit-email" type="text" id="edit-email" name="cm-ilnurl-ilnurl" value="" size="60" maxlength="128" placeholder="E-mailadres" class="form-text required form-control" required="required" aria-required="true" />

        </div>
    <button class="btn-lg button js-form-submit form-submit btn btn-secondary" data-drupal-selector="edit-submit" type="submit" id="edit-submit" name="op" value="Houd mij op de hoogte">Houd mij op de hoogte</button>
  <input data-drupal-selector="form-lnjdpsbx7hgh8d-5poj1ez5tk41ae8ulrhar2yubcsw" type="hidden" name="form_build_id" value="form-LnjdPSBX7hgH8D-5poJ1EZ5TK41Ae8ulrHar2YuBcsw" />
  <input data-drupal-selector="edit-promail-form" type="hidden" name="form_id" value="promail_form" />

</form>

  </div>
</div>
        </div>

  </div>
</div>
      

  </div>
  <div class="paragraph-text-three-cols__col3 paragraph-text-three-cols__col3--beige">
            
              <div class="paragraph-text-three-cols__col3-item"><h2 class="heading-lg">Kom kijken!</h2>

<p class="par-md"><strong>Wij laten je graag meer zien van ons werk en onze dieren.</strong></p>

<p class="par-md">Elke vrijdag, zaterdag en zondag verzorgt AAP unieke rondleidingen op het terrein. De minimum leeftijd hiervoor is 12 jaar, huisdieren zijn niet toegestaan. De rondleidingen zijn gratis en je kunt voor maximaal 2 personen tegelijk reserveren.</p>

<p><a class="btn btn-lg btn-secondary" href="/nl/form/tour" title="Een rondleiding bij AAP">Aanmelden voor een rondleiding</a></p></div>
      

    
  </div>
  <div class="lane__break"></div>
</div>
</div>
        </div>

  </div>

  
</article>

      </div>

                </div>
              </div>          </div>

                  </div>
      </div>

      
      
          </main>
  </div>

    <footer class="page__footer">
    <div class="container">
      <div class="row">
          <div class="page__footer-inner">
              <div class="region-footer">
          <div id="block-footersitemap" class="block-footersitemap">
  
    
      
              <ul class="block-footersitemap__main-topmenu">
                            <li class="block-footersitemap__main-topmenu-item">
            <a href="/nl/help-mee" data-drupal-link-system-path="node/3148">Help mee</a>
                                        <ul class="block-footersitemap__main-submenu block-footersitemap__main-topmenu-submenu">
                            <li class="block-footersitemap__main-submenu-item">
            <a href="/nl/help-mee/marria-donatie">Doneren</a>
                      </li>
                            <li class="block-footersitemap__main-submenu-item">
            <a href="/nl/help-mee/adopteer-een-dier" data-drupal-link-system-path="help-mee/adopteer-een-dier">Adoptie</a>
                      </li>
                            <li class="block-footersitemap__main-submenu-item">
            <a href="/nl/help-mee/air-miles-doneren-marria" title="Doneer je Air Miles voor dieren als Marria, kies eenmalig of jaarlijks">Air Miles doneren</a>
                      </li>
                            <li class="block-footersitemap__main-submenu-item">
            <a href="/marria">Mobieltjes en/of cartridges inleveren</a>
                      </li>
                            <li class="block-footersitemap__main-submenu-item">
            <a href="/nl/help-mee/toners-en-cartridges" data-drupal-link-system-path="node/1441">Cartridges inleveren met je bedrijf</a>
                      </li>
                            <li class="block-footersitemap__main-submenu-item">
            <a href="/nl/help-mee/aap-testament" data-drupal-link-system-path="node/2938">AAP in testament</a>
                      </li>
                            <li class="block-footersitemap__main-submenu-item">
            <a href="/nl/belastingvoordeel" data-drupal-link-system-path="node/1447">Geven met belastingvoordeel</a>
                      </li>
                            <li class="block-footersitemap__main-submenu-item">
            <a href="/nl/help-mee/help-mee-met-een-grote-gift" title="Doe een grote gift of steun een project">Help mee met een grote gift</a>
                      </li>
                            <li class="block-footersitemap__main-submenu-item">
            <a href="https://www.aap.nl/nl/help-mee/help-mee-met-uw-bedrijf">Steun AAP met uw bedrijf</a>
                      </li>
                            <li class="block-footersitemap__main-submenu-item">
            <a href="/nl/form/donation_box" title="Donatiebox voor bijv. verjaardag of bruiloft aanvragen" data-drupal-link-system-path="form/donation_box">Vraag een donatiebox aan</a>
                      </li>
                            <li class="block-footersitemap__main-submenu-item">
            <a href="/nl/help-mee/ons-verlanglijstje" data-drupal-link-system-path="node/3495">Ons verlanglijstje</a>
                      </li>
              </ul>
      
                      </li>
                            <li class="block-footersitemap__main-topmenu-item">
            <a href="/nl/dit-doen-wij" data-drupal-link-system-path="what-we-do">Dit doen wij</a>
                                        <ul class="block-footersitemap__main-submenu block-footersitemap__main-topmenu-submenu">
                            <li class="block-footersitemap__main-submenu-item">
            <a href="/nl/dit-doen-wij/opvang" data-drupal-link-system-path="node/2939">Opvang</a>
                      </li>
                            <li class="block-footersitemap__main-submenu-item">
            <a href="/nl/dit-doen-wij/rehabilitatie-en-verzorging" data-drupal-link-system-path="node/2945">Rehabilitatie</a>
                      </li>
                            <li class="block-footersitemap__main-submenu-item">
            <a href="/nl/dit-doen-wij/herplaatsing" data-drupal-link-system-path="node/2958">Herplaatsing</a>
                      </li>
                            <li class="block-footersitemap__main-submenu-item">
            <a href="/nl/dit-doen-wij/wetgeving-en-beleid" data-drupal-link-system-path="node/2962">Wetgeving en Beleid</a>
                      </li>
                            <li class="block-footersitemap__main-submenu-item">
            <a href="/nl/dit-doen-wij/2016-vogelvlucht" data-drupal-link-system-path="node/2799">Resultaten</a>
                      </li>
              </ul>
      
                      </li>
                            <li class="block-footersitemap__main-topmenu-item">
            <a href="/nl/kom-kijken" data-drupal-link-system-path="node/1443">Kom kijken</a>
                                        <ul class="block-footersitemap__main-submenu block-footersitemap__main-topmenu-submenu">
                            <li class="block-footersitemap__main-submenu-item">
            <a href="/nl/form/tour" data-drupal-link-system-path="form/tour">Aanmelden voor een rondleiding</a>
                      </li>
              </ul>
      
                      </li>
                            <li class="block-footersitemap__main-topmenu-item">
            <a href="/nl/actueel" data-drupal-link-system-path="news-and-blogs">Actueel</a>
                      </li>
                            <li class="block-footersitemap__main-topmenu-item">
            <a href="/nl/search" data-drupal-link-system-path="search">Zoeken</a>
                      </li>
              </ul>
      


              <ul class="block-footersitemap__top-topmenu">
                            <li class="block-footersitemap__top-topmenu-item">
            <a href="/nl/over-ons-0" data-drupal-link-system-path="node/3314">Over ons</a>
                                        <ul class="block-footersitemap__top-submenu block-footersitemap__top-topmenu-submenu">
                            <li class="block-footersitemap__top-submenu block-footersitemap__top-topmenu-submenu">
            <a href="/nl/over-ons/organisatie" data-drupal-link-system-path="node/1479">Organisatie</a>
                      </li>
                            <li class="block-footersitemap__top-submenu block-footersitemap__top-topmenu-submenu">
            <a href="/nl/over-ons/historie" data-drupal-link-system-path="node/1481">Historie</a>
                      </li>
                            <li class="block-footersitemap__top-submenu block-footersitemap__top-topmenu-submenu">
            <a href="/nl/over-ons/werken-bij" data-drupal-link-system-path="node/1485">Werken bij</a>
                      </li>
                            <li class="block-footersitemap__top-submenu block-footersitemap__top-topmenu-submenu">
            <a href="/nl/over-ons/werken-bij/stages-bij-stichting-aap" title="Stage lopen bij Stichting AAP">Stage lopen</a>
                      </li>
                            <li class="block-footersitemap__top-submenu block-footersitemap__top-topmenu-submenu">
            <a href="/nl/over-ons/werken-bij/vrijwilligerswerk" title="Vrijwilligerswerk bij AAP">Vrijwilligerswerk</a>
                      </li>
                            <li class="block-footersitemap__top-submenu block-footersitemap__top-topmenu-submenu">
            <a href="/nl/help-mee/bestaande-partners" data-drupal-link-system-path="node/1551">Partners</a>
                      </li>
              </ul>
      
                      </li>
                            <li class="block-footersitemap__top-topmenu-item">
            <a href="/nl/contact" data-drupal-link-system-path="node/2999">Contact</a>
                                        <ul class="block-footersitemap__top-submenu block-footersitemap__top-topmenu-submenu">
                            <li class="block-footersitemap__top-submenu block-footersitemap__top-topmenu-submenu">
            <a href="/nl/content/persvragen-en-contact" data-drupal-link-system-path="node/1706">Pers</a>
                      </li>
                            <li class="block-footersitemap__top-submenu block-footersitemap__top-topmenu-submenu">
            <a href="/nl/form/spreekbeurt_of_werkstuk" data-drupal-link-system-path="form/spreekbeurt_of_werkstuk">Spreekbeurt of werkstuk</a>
                      </li>
                            <li class="block-footersitemap__top-submenu block-footersitemap__top-topmenu-submenu">
            <a href="/nl/contact/veelgestelde-vragen-over-doneren" title="Veelgestelde vragen" data-drupal-link-system-path="node/1894">Veelgestelde vragen</a>
                      </li>
                            <li class="block-footersitemap__top-submenu block-footersitemap__top-topmenu-submenu">
            <a href="/nl/contact/ik-wil-mijn-dier-aanmelden" data-drupal-link-system-path="node/1523">Dier aanmelden</a>
                      </li>
                            <li class="block-footersitemap__top-submenu block-footersitemap__top-topmenu-submenu">
            <a href="/nl/contact/dierenleed-melden" data-drupal-link-system-path="node/1524">Meld dierenleed</a>
                      </li>
              </ul>
      
                      </li>
              </ul>
      


  </div>
<nav aria-labelledby="block-footermenunl-title" id="block-footermenunl" class="block-footermenunl">
            
  <h2 id="block-footermenunl-title" class="visually-hidden block-footermenunl__title">Footer menu (NL)</h2>
  

            

    
              <ul class="block-footermenunl__menu">
                            <li class="block-footermenunl__menu-item">
            <a href="/nl/user/login" class="block-footermenunl__menu-item-link" data-drupal-link-system-path="user/login">Login</a>
                    </li>
                            <li class="block-footermenunl__menu-item">
            <a href="/nl/colofon" class="block-footermenunl__menu-item-link" data-drupal-link-system-path="node/1492">Colofon</a>
                    </li>
                            <li class="block-footermenunl__menu-item">
            <a href="/nl/disclaimer-privacy" class="block-footermenunl__menu-item-link" data-drupal-link-system-path="node/1493">Disclaimer &amp; Privacy</a>
                    </li>
                            <li class="block-footermenunl__menu-item">
            <a href="/nl/cookies" class="block-footermenunl__menu-item-link" data-drupal-link-system-path="node/1494">Cookies</a>
                    </li>
              </ul>
      


  </nav>
<div id="block-socialprofilelinks" class="block-socialprofilelinks">
  
      <h2 class="block-socialprofilelinks__title">Volg Stichting AAP ook op:</h2>
    
      
  <div class="item-list-custom-social-profile-links">
      <ul class="social-profile-links item-list-custom-social-profile-links__list">
          <li class="social-profile-links__item social-profile-links__item--twitter item-list-custom-social-profile-links__list-item"><a href="http://www.twitter.com/stichting_aap" class="social-profile-links__item-link" target="_blank">Twitter</a></li>
          <li class="social-profile-links__item social-profile-links__item--pinterest item-list-custom-social-profile-links__list-item"><a href="http://www.pinterest.com/stichtingaap/" class="social-profile-links__item-link" target="_blank">Pinterest</a></li>
          <li class="social-profile-links__item social-profile-links__item--youtube item-list-custom-social-profile-links__list-item"><a href="http://www.youtube.com/user/StichtingAAP" class="social-profile-links__item-link" target="_blank">Youtube</a></li>
          <li class="social-profile-links__item social-profile-links__item--facebook item-list-custom-social-profile-links__list-item"><a href="http://www.facebook.com/StichtingAAP" class="social-profile-links__item-link" target="_blank">Facebook</a></li>
          <li class="social-profile-links__item social-profile-links__item--linkedin item-list-custom-social-profile-links__list-item"><a href="https://www.linkedin.com/company/aap" class="social-profile-links__item-link" target="_blank">Linkedin</a></li>
      </ul>
  </div>
  </div>

      </div>


            <div class="col-sm-6">
              <p class="align-right logo-container">
                <a href="https://www.aap.nl/nl/over-de-nationale-postcode-loterij">
                  <img src="/themes/aap/images/logos/npl.png?v2=" alt="Stichting AAP en de Postcodeloterij">
                </a>
                <img src="/themes/aap/images/logos/cbf-anbi.png" alt="logo of CBF and ANBI">
              </p>
            </div>
        </div>
      </div>
    </div>
  </footer>
  
</div>


<script type="application/json" data-drupal-selector="drupal-settings-json">{"path":{"baseUrl":"\/","scriptPath":null,"pathPrefix":"nl\/","currentPath":"node\/3667","currentPathIsAdmin":false,"isFront":true,"currentLanguage":"nl"},"pluralDelimiter":"\u0003","jquery":{"ui":{"datepicker":{"isRTL":false,"firstDay":1}}},"ajaxTrustedUrl":{"https:\/\/webgenerator.createsend.com\/t\/r\/s\/ilnurl\/":true},"user":{"uid":0,"permissionsHash":"007129e9d3c79e2ebef04bfca6a327c5aebab93019f6c07890ef28130e7f40e8"}}</script>

<!--[if lte IE 8]>
<script src="/uploads/js/js_5RH3jC1tW_hjk-Nh0Q_ZFhpL-scB_L3h1PfGSh7kitQ.js"></script>
<![endif]-->
<script src="/uploads/js/js_YnAZ6PVFg8jQshyxLHscIOl1gOfarmNv1uDRfrn8G_0.js"></script>
<script src="//use.typekit.net/vsp8csv.js"></script>
<script src="/uploads/js/js_bGsbb0dBhzUwHrbDzVnr1l1GNKunD0CQsQu_a_ta0h8.js"></script>

</body>
</html>
```

**Comment by JanWielemaker** [comment]
Thanks.  Fixed with d97af60fa294267f491f26dd78347142b8568e40.  The issue arises from the UTF-8 stream encoding, which causes the number of characters to be less then the number of bytes.  It now resizes the buffer to the next power of 2 until it is long enough to get enough data (or end-of-file).
